src/main/ruby/core/enumerable.rb

```diff
@@ -112,6 +112,7 @@ def count(item = undefined)
     seq = 0
     if !undefined.equal?(item)
       each { |o| seq += 1 if item == o }
+      warn 'warning: given block not used' if block_given?
```

Comment:
This is due to the Array literal being evaluated before the call to `count`.
The code below shows the warning is printed before calling each:
```ruby
[3] pry(main)> obj = Object.new
=> #<Object:0x00559bf4245a28>
[4] pry(main)> def obj.each; p :each; end
=> :each
[5] pry(main)> obj.extend Enumerable
=> #<Object:0x00559bf4245a28>
[6] pry(main)> obj.count(nil) { }
(pry):6: warning: given block not used
:each
```
or maybe more simply:
```ruby
[7] pry(main)> obj = Object.new
=> #<Object:0x00559bf3ad9230>
[8] pry(main)> obj.extend Enumerable
=> #<Object:0x00559bf3ad9230>
[9] pry(main)> obj.count(nil) { }
(pry):9: warning: given block not used
NoMethodError: undefined method `each' for #<Object:0x00559bf3ad9230>
```